### PR TITLE
Add Reverse lookup "transfer only" mode

### LINKED ISSUE #107: Add OwnerReverseLookupMode::TransfersOnly
Add OwnerReverseLookupMode::TransfersOnly option for usecases where the majority of NFTs are owned by the private minter and only NFT's that have been transferred benefit from reverse lookup tracking (edited) 

https://casperlabs-team.slack.com/archives/C02923Q9QSC/p1671566313301129

### PR DESCRIPTION
**Pull request by gRoussac**
Add TransfersOnly into OwnerReverseLookupMode with corresponding **tests** in installer/mint/burn/transfer/costs

### CODE REVIEW

File: contract/src/main.rs [hunk 1]
```diff
@@ -404,6 +416,50 @@ pub extern "C" fn set_variables() {
     }
 }
 
+fn add_page_entry_and_page_record(tokens_count: u64, item_key: &str) -> (u64, URef) {
```
Comment (darthsiroftardis) [inline contract/src/main.rs]: Lets move this function out into our utils module

File: contract/src/main.rs [hunk 2]
```diff
@@ -830,6 +847,59 @@ pub extern "C" fn set_approval_for_all() {
     }
 }
 
+fn update_page_entry_and_page_record(
```
Comment (darthsiroftardis) [inline contract/src/main.rs]: lets move this out to utils as well

File: contract/src/main.rs (hunk 1 above)
Comment (gRoussac) [reply contract/src/main.rs]: Moved thanks

File: contract/src/main.rs (hunk 2 above)
Comment (gRoussac) [reply contract/src/main.rs]: moved as well thanks

File: contract/src/error.rs [hunk 3]
```diff
@@ -129,8 +129,8 @@ pub enum NFTCoreError {
     MissingMigrationFlag = 124,
     InvalidMigrationFlag = 125,
     ContractAlreadyMigrated = 126,
-    UnregisteredOwnerInMint = 127,
-    UnregisteredOwnerInTransfer = 128,
+    UnregisteredOwnerInPageAdd = 127,
```
Comment (darthsiroftardis) [inline contract/src/error.rs]: Lets keep the original names of the errors, although its an error code, I'd like to keep the names consistent from minor release

Comment (gRoussac) [reply contract/src/error.rs]: @darthsiroftardis I get the consistency matter but the fact is that `UnregisteredOwnerInMint` could then be triggered into a `Transfer` and that seems to me confusing for the user.

Shall I add two new errors, and `UnregisteredOwnerInMint` / `UnregisteredOwnerInTransfer` get deprecated ?

**Review by darthsiroftardis — APPROVED** [review]
Approval pending updating the README describing the new modality

**Comment by gRoussac** [comment]
@darthsiroftardis Please have a look at https://github.com/casper-ecosystem/cep-78-enhanced-nft/pull/144/commits/f48e9f1e7e381f446dfa7728b76ec577c63c72e5 let me know if this is correct English and clear enough thanks

File: contract/src/error.rs (hunk 3 above)
Comment (gRoussac) [reply contract/src/error.rs]: @darthsiroftardis As suggested let me know what you think of this revert and flag thanks!

https://github.com/casper-ecosystem/cep-78-enhanced-nft/pull/144/commits/d9456b97be9fe07d7c4e6aeb3d1d50745ff8e524

Comment (gRoussac) [reply contract/src/error.rs]: made a simple flag true/false

**Review by deuszex — APPROVED** [review]

**Pull request merged**